src/main/java/taskmaster/tasks/TaskList.java

```diff
@@ -0,0 +1,241 @@
+     */
+    public void addTask(TaskType taskType, String description, String marked) throws DukeException {
+        if (taskType == TaskType.TODO) {
+            if (description.isEmpty()) {
+                throw new DukeException("The description of a todo cannot be empty.");
+            }
+            list.add(new Todo(description, marked));
+            System.out.println(Ui.line);
+            System.out.println("Got it. I've added this to-do task:");
+            System.out.println("  " + list.get(list.size() - 1));
+            System.out.println("Now you have " + list.size() + " tasks in the list.");
+            System.out.println(Ui.line);
+        } else if (taskType == TaskType.EVENT) {
+            boolean wrongInput = false;
+            String[] parts = description.split("/from");
+            if (parts.length == 2) {
+                String details = parts[0].trim();
+                String[] timeParts = parts[1].split("/to");
+                if (timeParts.length == 2) {
+                    String from = timeParts[0].trim();
+                    String end = timeParts[1].trim();
+                    list.add(new Event(details, from, end, marked));
+                } else {
+                    wrongInput = true;
+                }
+            } else {
+                wrongInput = true;
+            }
+            if (wrongInput) {
+                throw new DukeException("Please input a valid task");
+            } else {
+                System.out.println(Ui.line);
+                System.out.println("Got it. I've added this event:");
+                System.out.println("  " + list.get(list.size() - 1));
+                System.out.println("Now you have " + list.size() + " tasks in the list.");
+                System.out.println(Ui.line);
+            }
+        } else if (taskType == TaskType.DEADLINE) {
+            boolean wrongInput = false;
+            String[] parts = description.split("/by");
+            if (parts.length == 2) {
+                String details = parts[0].trim();
+                String by = parts[1].trim();
+                list.add(new Deadline(details, by, marked));
+            } else {
+                wrongInput = true;
+            }
+            if (wrongInput) {
+                throw new DukeException("Please input a valid task");
+            } else {
+                System.out.println(Ui.line);
+                System.out.println("Got it. I've added this deadline:");
+                System.out.println("  " + list.get(list.size() - 1));
+                System.out.println("Now you have " + list.size() + " tasks in the list.");
+                System.out.println(Ui.line);
+            }
+        }
+    }
+    /**
+     * Prints all tasks that is happening on the specified date.
+     *
+     * @param date Date that task is happening on.
+     */
+    public void printTasksByDate(String date) {
+        LocalDate dueDate = null;
```

Comment:
Maybe this should have a modifier? Might want to use `private` if you're able to.